### DISCUSSION TITLE
Next.js | favicon | best practice

### DISCUSSION THREAD

**Discussion by enkicoma** [vercel/next.js#21496] [Help] [answered]
![Screenshot 2021-01-24 at 19 47 01](https://user-images.githubusercontent.com/29044574/105641637-04d62c80-5e7d-11eb-9c01-2f2f87f38052.jpg)

Hi guys,

regarding the above screenshot.

How Next.js does this favicon? I have one `favicon.ico` file in my `public` folder and Next.js picked up it automatically... If I will include the rest files:
1. will it be picked up? 
2. or if I want to include them in a different path.. how can I do it?

**Comment by enkicoma** [comment, marked as answer]
found it...
will put it in a head
`import Head from 'next/head';`
```js
       <Head>
          <link rel="apple-touch-icon" sizes="180x180" href="/favicons/apple-touch-icon.png" />
          <link rel="icon" type="image/png" sizes="32x32" href="/favicons/favicon-32x32.png" />
          <link rel="icon" type="image/png" sizes="16x16" href="/favicons/favicon-16x16.png" />
          <link rel="manifest" href="/favicons/site.webmanifest" />
          <link rel="mask-icon" href="/favicons/safari-pinned-tab.svg" color="#5bbad5" />
          <meta name="msapplication-TileColor" content="#ffc40d" />
          <meta name="theme-color" content="#ffffff" />
        </Head>
        
      ```

**Comment by agusterodin** [comment]
I wish they had favicon generator build-in of the box as well, something similar to gatsby-plugin-manifest. In the meantime here are two solutions:

https://github.com/agusterodin/Modern-Favicons (made by me, very lightweight)
https://github.com/tinialabs/next-favicon-loader (more of a gatsby-plugin-manifest equivalent)

**Reply by moritzjacobs** [reply]
I just used your CLI tool, very nice! Thanks!

**Reply by ChocolateLoverRaj** [reply]
@agusterodin Why does Modern-Favicons not have issues on GitHub?

**Reply by agusterodin** [reply]
To protect my mental health.

I would encourage anyone who wishes to extend/improve it to create their own fork.

**Reply by moritzjacobs** [reply]
> To protect my mental health.

Wise words!
